library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -994,6 +1001,9 @@ public int getMaxRetryCount()
    */
   public void setMaxRetryCount(int maxRetryCount)
   {
+    if (maxRetryCount < 0) {
+      throw new IllegalArgumentException("maxRetryCount should be greater than or equal to 0.");
+    }
```

Comment:
@tweise @chaithu14
I took the change in https://github.com/apache/apex-malhar/blob/master/library/src/main/java/org/apache/apex/malhar/lib/fs/FSRecordReader.java#L158

I had tried the \@Min constraint annotation, which throws a constraint violation exception if the values are invalid in the properties.xml (bean validation). However, I do not see it throwing exceptions during the setting of the property of the operator when the application is running. Please let me know if I have misunderstood something here.